wremnants/muon_selections.py

```diff
@@ -56,12 +56,17 @@ def select_veto_muons(df, nMuons=1, condition="==", ptCut=10.0, etaCut=2.4):
 
 '''
 
-def select_veto_muons(df, nMuons=1, condition="==", ptCut=15.0, etaCut=2.4):
+def select_veto_muons(df, nMuons=1, condition="==", ptCut=15.0, staPtCut=15.0, etaCut=2.4, useGlobalOrTrackerVeto=False):
 
     # n.b. charge = -99 is a placeholder for invalid track refit/corrections (mostly just from tracks below
     # the pt threshold of 8 GeV in the nano production)
     df = df.Define("vetoMuonsPre", "Muon_looseId && abs(Muon_dxybs) < 0.05 && Muon_correctedCharge != -99")
-    df = df.Define("vetoMuonsPre2", "vetoMuonsPre && Muon_isGlobal && Muon_highPurity && Muon_standalonePt > 15 && Muon_standaloneNumberOfValidHits > 0 && wrem::vectDeltaR2(Muon_standaloneEta, Muon_standalonePhi, Muon_correctedEta, Muon_correctedPhi) < 0.09")
+    df = df.Define("Muon_isGoodGlobal", f"Muon_isGlobal && Muon_highPurity && Muon_standalonePt > {staPtCut} && Muon_standaloneNumberOfValidHits > 0 && wrem::vectDeltaR2(Muon_standaloneEta, Muon_standalonePhi, Muon_correctedEta, Muon_correctedPhi) < 0.09")
```

Comment:
This is actually fine, but I would still like to propose to do it something like in this function [0], where there is also the possibility to use the original definition (even if for that one there would be no SF available), for tests or whatever.

Aside from that, if you keep your code I would swap the "if not useGlobalOrTrackerVeto:" conditions so to use directly "if useGlobalOrTrackerVeto", it seems more natural to me to avoid one negation in this context


[0] https://github.com/cippy/WRemnants/blob/wlikeIso/wremnants/muon_selections.py#L46-L62

Reply:
In the end I just swapped the condition